### ISSUE TITLE
Closing DSSI-VST synth gui crashes Muse

### ISSUE THREAD

**Issue by asbaklm** [muse-sequencer/muse#382]
Experiencing Muse crashes when closing GUI for DSSI-VST synths

ERROR: Remote VST plugin communication failure in GUI thread
Segmentation fault
Remote VST plugin audio thread: returning
fixme:win:UnregisterDeviceNotification (handle=0xcafeaffe), STUB!

**Comment by danvd** [comment]
Hi @asbaklm,

The behavior of dssi-vst highly depends on win32 vst plugin used. Can you post this info too?

Regards,
Andrew.

**Comment by asbaklm** [comment]
Hi there, this happens with both PG-8X and Kontakt. I haven't checked with other plugins but assumed they'd also crash Muse when closing the GUI.

**Comment by spamatica** [comment]
Hi,

DSSI-VST, being based on wine is a bit of a moving target I'm afraid.
I'm struggling myself at the moment, to get the kx-studio version of
DSSI-VST to work correctly, jack goes nuts due to a timeout induced by
DSSI-VST.

Could you try some other DSSI capable host and perform the same procedure,
see if it misbehaves there too?

We probably need more information about versions of stuff,  DSSI-VST, wine,
muse, your distribution, etc.

Regards,
Robert

Den 26 okt 2015 19:11 skrev "asbaklm" <notifications@github.com>:
>
> Hi there, this happens with both PG-8X and Kontakt. I haven't checked
with other plugins but assumed they'd also crash Muse when closing the GUI.

**Comment by asbaklm** [comment]
Thanks for the information! I just had a thought and realised that on one computer I have 64bit wine and on another 32bit wine. It looks like there is no crashing on the 32bit wine system with jack2. Perhaps it's just coincidence, perhaps not. 

Perhaps it's better to stick with 32bit wine, generally speaking.

**Comment by asbaklm** [comment]
I think I may have found the problem, and I don't think it was an issue with wine. 

Rather, it seems to be related to airwave vst bridge. If this is installed and a link is created for a Windows vsti (it basically seems to create a virtual Linux vst) then Windows vsti's (loaded via dssi-vst) crash Muse once the GUI is closed. 

Also, when trying to load the "native" bridged VST in Muse crashes it immediately. Airwave seemed to work in some other hosts like Carla.

After removing the airwave links and uninstalling it, no more Windows vsti crashes in Muse with dssi-vst. I'll leave a note for the developer of airwave. Many thanks!

**Comment by spamatica** [comment]
Hi,

I wasn't aware of Airwave, I'll have to give it a try, or does it not work
with MusE?

As for carla working better, well dssi-vst is quite a bit older so it's not
not so strange if there are more issues.
One of these these days I hope we can add a new implementation for Wine
VSTs, thanks for mentioning the Airwave in that regard.

Regards,
Robert


2015-10-28 20:12 GMT+01:00 asbaklm <notifications@github.com>:

**Comment by asbaklm** [comment]
My experiences (perhaps it's different for others):

- Airwave does not seem to work with Muse. (Or Ardour, unless it is loaded via a Carla plug in Ardour). When trying to load a "virtual" (bridged) Windows VSTI, Аirwave immediately crashes Muse. Unfortunately I don't code so don't know where the issue between Muse and Airwave seems to be or how to resolve it. But please try it for yourself just to gain some insight into the issue, because Airwave does seem like a promising tool to host Windows VSTIs.

- DSSI-VST plugs hosted in Muse crash while Airwave is installed on the computer when closing the VSTI GUI, (even though Airwave isn't being used to host the VSTI). To resolve the crashing issue I had to purge Airwave from my system after which DSSI-VSTI plugs behaved normally again in Muse.

- DSSI-VST provides a reasonably reliable, low-latency and xrun free Windows VSTI experience, it seems to be more reliable than some of the other solutions. In Muse it seems to be the best current solution. (Unless there's something else out there that I'm not aware of).

- Using a Windows VSTI host like Herman Seib's Savihost also works reasonably well for hosting Windows VSTIs, but it's obviously not integrated into Muse itself.

- Carla is a good host for LV2, Ladspa, Linux VST's.

but

- Carla is not the most reliable Windows VSTI host, carla xruns a lot more than DSSI-VST so it's basically unusable in its current state for hosting Windows VSTI's. (Unless the user didn't care about running with high latencies).

So basically, I think that Airwave shows a lot of promise but in its current state there seems to be some operability issues which first need resolving.
